Kernel/Prekernel/init.cpp

```diff
@@ -253,6 +292,29 @@ u64 generate_secure_seed()
     return seed;
 }
 
+void* multiboot2_find_next_tag(u8* data, u32 type)
+{
+    u8* current = data;
+    auto tag = (multiboot2_tag_t*)current;
+
+    while (1) {
+        if (tag->type == MULTIBOOT2_TAG_TYPE_END)
+            return nullptr;
+        if (tag->type == type)
+            return tag;
+
+        current += tag->size;
+        while ((uintptr_t)current & 7)
+            current++;
+        tag = (multiboot2_tag_t*)current;
```

Comment:
Overall remark: can we use C++-style `static_cast`s/`reinterpret_cast`s where applicable?

Reply:
You're absolutely right; used too much typescript lately I suppose. Will fix.

Reply:
> Overall remark: can we use C++-style `static_cast`s/`reinterpret_cast`s where applicable?

Fixed!